### ISSUE TITLE
Discussion: Supporting OTLP protocol directly

### ISSUE THREAD

**Issue by hlinnaka** [DataDog/pg_tracing#9]
## Problem

Currently, you need another program to poll the `pg_tracing_consume_spans` view and push the spans to an OpenTelemetry Collector. See https://github.com/bonnefoa/pg-tracing-otel-forwarder. I wish the extension could connect directly to the OpenTelemetry Collector, using the OTLP protocol. That would simplify the deployment.

This was discussed on pgsql-hackers: https://www.postgresql.org/message-id/CAO6_Xqr5cNC-fK7kX4Pt9LkYSNjuZQjDfxbm_ckypM9LD8PT1Q%40mail.gmail.com. A pull model made sense for a patch for core PostgreSQL, to avoid having a dependency on HTTP or gRPC or other heavy libraries. But as an extension, that's less of an issue.

## Alternatives

One approach is to implement the simplest OTLP/HTTP in JSON protocol using libcurl. Libcurl is widely available and doesn't require a lot of extra baggage.

Another approach would be to add a dependency to the OpenTelemetry C++ client library (https://github.com/open-telemetry/opentelemetry-cpp). That would come with more full support for different transports, but is a much bigger dependency.

Yet another approach would be to bundle a standalone binary like pg-tracing-otel-forwarder with this extensions. The extension could launch it on startup as a separate process, for example. Or just improve the binary to make it nicer to deploy, even though it needs to be deployed separately.

I'm leaning towards the libcurl approach myself. What do you think? 

If we go down this route, we could simplify the rest of the code by removing the pull model. It's nice to have *something* to play with, though, even when you don't have OpenTelemetry colletor or something like Jaeger running. Maybe add an option to dump all the spans to a file instead of pushing them with OTLP?

**Comment by bonnefoa** [comment]
> I'm leaning towards the libcurl approach myself. What do you think?

I would definitely prefer this over adding opentelemetry-cpp as a dependency. I took a stab at implementing it in https://github.com/DataDog/pg_tracing/pull/27 and was able to send json spans with libcurl through OTLP/HTTP. 

It currently uses a background worker that sends the spans to a defined otel collector at regular intervals (function to trigger sending spans or sending spans when buffer is x% full could be added later on). This allows to keep curl usage in a dedicated process and simplify sharing a single connection to the otel exporter for the whole db.

> It's nice to have something to play with, though, even when you don't have OpenTelemetry colletor

I've added the `select pg_tracing_json_spans()` function to output the json that would be sent to the otel collector. It is also used for testing. Most of my tests still rely very heavily on the `pg_tracing_{peek,consume}_spans` but it would be possible to convert them to rely on the json export instead.
